packages/webvh/src/dids/didWebvhUtil.ts

```diff
@@ -0,0 +1,119 @@
+    try {
+      const kms = this.agentContext.dependencyManager.resolve(Kms.KeyManagementApi)
+
+      const publicJwk = Kms.PublicJwk.fromPublicKey({
+        kty: 'OKP',
+        crv: 'Ed25519',
+        publicKey: publicKey,
+      })
+
+      const verificationResult = await kms.verify({
+        key: { publicJwk: publicJwk.toJson() },
+        algorithm: 'EdDSA',
+        signature: signature,
+        data: message
+      })
+
+      return verificationResult.verified
+    } catch (error) {
+      this.agentContext.config.logger.error('KMS verification failed:', error)
+      return false
+    }
+  }
+
+  private async verifyWithLegacyMethod(signature: Uint8Array, message: Uint8Array, publicKey: Uint8Array): Promise<boolean> {
+    try {
+      // Fallback for older versions of Credo-TS without KMS
+      // Use @stablelib/ed25519 for Ed25519 verification
+      try {
+        const { verify } = await import('@stablelib/ed25519')
+        
+        this.agentContext.config.logger.debug('Using @stablelib/ed25519 for signature verification')
+        
+        if (publicKey.length !== 32) {
+          this.agentContext.config.logger.error(`Invalid public key length: expected 32 bytes, got ${publicKey.length}`)
+          return false
+        }
+        
+        if (signature.length !== 64) {
+          this.agentContext.config.logger.error(`Invalid signature length: expected 64 bytes, got ${signature.length}`)
+          return false
+        }
+        return verify(publicKey, message, signature)
+      } catch (importError) {
+        this.agentContext.config.logger.error('Failed to import @stablelib/ed25519:', importError)
+        this.agentContext.config.logger.warn('No crypto utilities available for signature verification')
+        return false
+      }
+    } catch (error) {
+      this.agentContext.config.logger.error('Legacy verification failed:', error)
+      return false
+    }
+  }
+
+  public async verify(signature: Uint8Array, message: Uint8Array, publicKey: Uint8Array): Promise<boolean> {
+    try {
+      if (!this.agentContext) {
+        throw new Error('Agent context is required')
+      }
+
+      // Check if KMS is available in this version of Credo-TS
+      const kmsAvailable = await this.isKmsAvailable()
+      
+      if (kmsAvailable) {
+        this.agentContext.config.logger.debug('Using KMS for signature verification')
+        return await this.verifyWithKms(signature, message, publicKey)
+      } else {
+        this.agentContext.config.logger.debug('KMS not available, using legacy verification method')
+        return await this.verifyWithLegacyMethod(signature, message, publicKey)
+      }
+    } catch (error) {
+      // Log error in a non-production environment
+      if (process.env.NODE_ENV !== 'production') {
+        this.agentContext.config.logger.error('Error verifying signature:', error)
+      }
+      return false
+    }
+  }
+}
+
+export interface IDidDocOptions {
```

Comment:
Where is this interface used? We generally don't use `I` prefix for interfaces, and also write out the full name, so `DidDocumentOptions`, but might be good to make it more specific if it's related to WebVh.